### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by adamnemecek**

### CODE REVIEW

File: WWDC/SessionsSplitViewController.swift [hunk 1]
```diff
@@ -27,13 +27,17 @@ final class SessionsSplitViewController: NSSplitViewController {
...
-        NotificationCenter.default.addObserver(self, selector: #selector(syncSplitView(notification:)), name: .sideBarSizeSyncNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(syncSplitView), name: .SideBarSizeSync, object: nil)
     }
 
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
+    deinit {
+        NotificationCenter.default.removeObserver(self)
```
Comment (allenhumphreys) [inline WWDC/SessionsSplitViewController.swift]: From apple documentation
> If your app targets iOS 9.0 and later or macOS 10.11 and later, you don't need to unregister an observer in its dealloc method.

As we are targeting macOS 10.12, these global removals on deinit are superfluous.

Comment (adamnemecek) [reply WWDC/SessionsSplitViewController.swift]: An ok, good to know. Those are weird ass semantics.

**Comment by allenhumphreys** [comment]
@adamnemecek Could you close the PR until you're done and it is ready for review. When the PR is open, some of us get emails every time you push a commit and it also triggers builds in travis.

**Comment by adamnemecek** [comment]
Ok let's say it's done.

File: ConfCore/UserDataSyncEngine.swift [hunk 2]
```diff
@@ -442,13 +442,13 @@ public final class UserDataSyncEngine {
         os_log("Will commit %{public}d changed record(s) to the database", log: log, type: .info, records.count)
 
         performRealmOperations { realm in
-            records.forEach { record in
```
Comment (allenhumphreys) [inline ConfCore/UserDataSyncEngine.swift]: screen shot 2018-06-11 at 2 17 08 pm

File: PlayerUI/Util/NSWindow+Snapshot.swift [hunk 3]
```diff
@@ -9,7 +9,6 @@
 import Cocoa
 
 extension NSWindow {
-
```
Comment (allenhumphreys) [inline PlayerUI/Util/NSWindow+Snapshot.swift]: I disagree with many removals of the vertical whitespace. Vertical whitespace can greatly improve readability.

File: PlayerUI/Util/NSWindow+Snapshot.swift [hunk 4]
```diff
@@ -18,9 +17,7 @@ extension NSWindow {
         }
 
         let image = NSImage(cgImage: cgImage, size: frame.size)
-
         image.cacheMode = .never
-
```
Comment (allenhumphreys) [inline PlayerUI/Util/NSWindow+Snapshot.swift]: Personally I prefer a newline before the return statement to make it clear the scope is ending.

File: WWDC/NSImage+Thumbnail.swift [hunk 5]
```diff
@@ -8,20 +8,40 @@
...
+extension NSRect {
+    init(width: CGFloat, height: CGFloat) {
+        self.init(x: 0, y: 0, width: width, height: height)
+    }
+
+    init(size: NSSize) {
+        self.init(origin: .zero, size: size)
+    }
+
+    var aspectRation: CGFloat {
```
Comment (allenhumphreys) [inline WWDC/NSImage+Thumbnail.swift]: Typo. Also, I'm not seeing where this is used.

File: WWDC/SessionCollectionViewItem.swift [hunk 6]
```diff
@@ -44,14 +44,10 @@ final class SessionCollectionViewItem: NSCollectionViewItem {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
```
Comment (allenhumphreys) [inline WWDC/SessionCollectionViewItem.swift]: I prefer a vertical whitespace separating the call to super from the rest of the function implementation.

File: WWDC/WWDCLayer.swift [hunk 7]
```diff
@@ -9,17 +9,13 @@
...
     }
-
 }
 
 final class WWDCShapeLayer: CAShapeLayer {
-
     override func action(forKey event: String) -> CAAction? {
         return nil
     }
-
```
Comment (allenhumphreys) [inline WWDC/WWDCLayer.swift]: This whitespace removal is the only one I agree with, but I think other contributors like it.

File: ConfCore/UserDataSyncEngine.swift (hunk 2 above)
Comment (adamnemecek) [reply ConfCore/UserDataSyncEngine.swift]: what version are you using?

Comment (allenhumphreys) [reply ConfCore/UserDataSyncEngine.swift]: Xcode 9.4. This file doesn't get compiled unless you're on an iCloud scheme

**Comment by insidegui** [comment]
As per the new contribution guidelines, we're not accepting PRs that only consist of code changes without bug fixes or significant improvements. I understand that the update to the guidelines was made after this PR was opened, but it was something we've been wanting to do for a long time. I'm sorry but I can't merge this.

**Pull request closed without merging**